tests/test_PsfFlux.py

```diff
@@ -100,11 +121,20 @@ def testSubImage(self):
         bbox = psfImage.getBBox()
         bbox.grow(-1)
         subExposure = exposure.Factory(exposure, bbox, lsst.afw.image.LOCAL)
+
         algorithm.measure(record, subExposure)
         self.assertFloatsAlmostEqual(record.get("base_PsfFlux_instFlux"), record.get("truth_instFlux"),
                                      atol=3*record.get("base_PsfFlux_instFluxErr"))
         self.assertTrue(record.get("base_PsfFlux_flag_edge"))
 
+        # Expected chi2 requires trimming PSF image by one pixel per side
+        # to match subExposure created above, so we can't use compute_chi2() directly here
+        scaledPsf = psfImage.array[1:-1, 1:-1]*record.get("base_PsfFlux_instFlux")
```

Comment:
Why do you need to trim the image, and why can't `compute_chi2` be designed to handle this case?

Reply:
Since `compute_chi2()` here is only to support these tests, I didn't want to figure out how to make it support this other use case.

The comment just above this line doesn't help you with the "why" question?

Reply:
Not really. You don't normally need to match anything, do you?

Reply:
The C++ code handles matching the PSF image and exposure image dimensions automatically. I didn't write the python code above to do the same, and just copied the relevant bits here. I tweaked the comment, not sure if it helps:

```
        # Calculating chi2 requires trimming the PSF image by one pixel per side
        # to match the subExposure created above, so we can't use compute_chi2()
        # directly here.
```